### PR TITLE
systemd-coredump: allow setting external core size to infinity

### PR DESCRIPTION
**Pull request by tblume**
Make it compatible to: ulimit -c unlimited

### CODE REVIEW

File: src/coredump/coredump.c [hunk 1]
```diff
@@ -157,7 +157,7 @@ static int parse_config(void) {
                 { "Coredump", "Storage",          config_parse_coredump_storage,  0, &arg_storage           },
                 { "Coredump", "Compress",         config_parse_bool,              0, &arg_compress          },
                 { "Coredump", "ProcessSizeMax",   config_parse_iec_uint64,        0, &arg_process_size_max  },
-                { "Coredump", "ExternalSizeMax",  config_parse_iec_uint64,        0, &arg_external_size_max },
+                { "Coredump", "ExternalSizeMax",  config_parse_iec_uint64_infinity,        0, &arg_external_size_max },
```
Comment (poettering) [inline src/coredump/coredump.c:160]: please realign the table

File: src/shared/conf-parser.c [hunk 2]
```diff
@@ -677,6 +677,33 @@ int config_parse_iec_uint64(
...
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (streq(rvalue, "infinity"))
+                return UINT64_MAX;
+
+        return config_parse_iec_uint64(unit, filename, line, section, section_line, lvalue
+                                       ltype, rvalue, data, userdata,);
```
Comment (poettering) [inline src/shared/conf-parser.c]: trailing comma?

File: src/shared/conf-parser.c [hunk 3]
```diff
@@ -677,6 +677,33 @@ int config_parse_iec_uint64(
...
+        uint64_t *bytes = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (streq(rvalue, "infinity"))
+                return UINT64_MAX;
```
Comment (poettering) [inline src/shared/conf-parser.c]: the value should be written into `*bytes`, not returned...

**Comment by poettering** [comment]
this certainly wasn't even compile tested.

also, needs man page update

**Comment by tblume** [comment]
> this certainly wasn't even compile tested.

Yes, sorry, that was rather for asking whether the approach is acceptable.
Will update the pr.

File: src/shared/conf-parser.c [hunk 4]
```diff
@@ -677,6 +677,30 @@ int config_parse_iec_uint64(
...
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint64_t *bytes;
+
+        assert(rvalue);
+
+        if (streq(rvalue, "infinity")) {
+                parse_size(UINT64_MAX, 1024, bytes);
```
Comment (poettering) [inline src/shared/conf-parser.c]: parse_size() accepts a string as first argument.

Just do:

```c
*bytes = UINT64_MAX;
return 0;
```

File: src/shared/conf-parser.c [hunk 5]
```diff
@@ -677,6 +677,30 @@ int config_parse_iec_uint64(
...
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint64_t *bytes = data;
+
+        assert(rvalue);
```
Comment (bluca) [inline src/shared/conf-parser.c:694]: Assert also on bytes or data

**Review by bluca — APPROVED** [review]

**Pull request merged**